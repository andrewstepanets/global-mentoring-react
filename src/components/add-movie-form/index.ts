export * from './add-movie-form';
