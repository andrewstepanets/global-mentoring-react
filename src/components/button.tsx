import React from "react";

export function Button({ symbol, onClickFunction }) {
  return <button onClick={onClickFunction}>{symbol}</button>;
}
