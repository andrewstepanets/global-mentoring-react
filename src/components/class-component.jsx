import React from "react";
class ClassComponent extends React.Component {
  render() {
    return (
      <div>
        <h2>Class Component</h2>
        <p>It's a Class Component</p>
      </div>
    );
  }
}

export default ClassComponent;
