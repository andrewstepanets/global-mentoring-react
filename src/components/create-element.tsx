import React from "react";

const text = "It's a component created by createElement";
export const h1 = React.createElement("h2", null, "Create Element");
export const p = React.createElement(
  "p",
  {
    id: "info-text",
    className: "details",
  },
  text
);
