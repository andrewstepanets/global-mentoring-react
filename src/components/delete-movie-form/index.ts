export * from './delete-movie-form';
