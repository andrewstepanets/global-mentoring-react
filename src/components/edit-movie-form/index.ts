export * from './edit-movie-form';
