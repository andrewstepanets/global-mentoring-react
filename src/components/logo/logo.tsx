import * as React from 'react';
import { Logo as LogoWrapper } from './styles';

export const Logo: React.FC = () => (
  <LogoWrapper to="/">
    <h2>
      netflix<span>roulette</span>
    </h2>
  </LogoWrapper>
);
