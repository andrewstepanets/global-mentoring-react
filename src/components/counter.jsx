import React, { useState } from "react";
import { Button } from "./button";

function Counter() {
  const [count, setCount] = useState(0);
  const incrementCount = () => setCount(count + 1);
  const decrementCount = () => setCount(count - 1);
  return (
    <div className="counter">
      <Button symbol="-" onClickFunction={decrementCount} />
      {count}
      <Button symbol="+" onClickFunction={incrementCount} />
    </div>
  );
}

export default Counter;
