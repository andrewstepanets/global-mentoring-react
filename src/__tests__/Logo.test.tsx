import * as React from 'react';
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';

import { Logo } from '../components/logo';

describe('<Logo />', () => {
  it('matches with snapshot', () => {
    const { baseElement } = render(
      <Router>
        <Logo />
      </Router>,
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('shows text', () => {
    const logoComponent = render(
      <Router>
        <Logo />
      </Router>,
    );
    const logoText = logoComponent.getByRole('heading', {
      name: /netflix roulette/i,
    });

    expect(logoText).toBeInTheDocument();
  });
});
