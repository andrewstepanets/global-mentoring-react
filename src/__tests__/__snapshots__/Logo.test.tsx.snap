// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> matches with snapshot 1`] = `
<body>
  <div>
    <a
      class="sc-bdvvtL emFNGz"
      href="/"
    >
      <h2>
        netflix
        <span>
          roulette
        </span>
      </h2>
    </a>
  </div>
</body>
`;
